rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java

```diff
@@ -480,7 +480,13 @@ private int doEnd(boolean cancel) {
                     status = toLRAStatus(status());
                 } else {
                     // forget calls for nested participants
-                    forgetAllParticipants();
+                    if (forgetAllParticipants()) {
+                        updateState(LRAStatus.Closed);
+                        return ActionStatus.COMMITTED;
```

Comment:
Do we need to call `runPostLRAActions();`. Have you tested your changes with an AfterLRAListener registered with the nested LRA?

Reply:
AfterLRA notifications should depend on the outcome of forget calls, as nested LRA is not finished before forget calls are delivered. Or did I misunderstood the scenario? I can add such test if needed.

Reply:
After your code changes the transaction end logic finishes early and you are skipping a section of code that we currently execute. In particular the code on line 
https://github.com/jbosstm/narayana/blob/0351d68661591df1fd53bd81d741fc7ded362bfb/rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java#L520 that was designed to perform the AfterLRA notifications is skipped.

Reply:
sorry @mmusgrov I've just got back to this. So if I remove this return statement then it should be ok right?

Reply:
@mmusgrov actually, it doesn't seem to matter to return here. I've added the test as you've requested.

Reply:
I am still uncomfortable why this route through the end() routine is returning early. No other code path returns early so it makes maintaining this method more difficult. We either need to avoid the early return or fully comment why it is safe to do so.

Thanks for the extra test - I didn't really understand what is was doing so some commentary would have been helpful to the both the reviewer and to future maintainers.